### PR TITLE
Add news endpoints

### PR DESCRIPTION
**Pull request by gnvk**

### CODE REVIEW

File: marketdata/entities.go [hunk 1]
```diff
@@ -211,6 +211,25 @@ type CryptoSnapshot struct {
...
 }
 
+// NewsImage is a single image for a news article.
+type NewsImage struct {
+	Size string `json:"size"`
+	URL  string `json:"url"`
+}
+
+// News is a single news article.
+type News struct {
```
Comment (kalmant) [inline marketdata/entities.go:221]: No ID?

File: marketdata/rest.go [hunk 2]
```diff
@@ -1234,6 +1239,108 @@ func (c *client) GetCryptoSnapshot(symbol string, exchange string) (*CryptoSnaps
...
+
+// List of sort values
+var (
+	// SortDesc means the results will be sorted in a descending order
+	SortDesc Sort = "desc"
+	// SortAsc means the results will be sorted in an ascending order
+	SortAsc Sort = "asc"
+)
+
+// GetNewsParams contains optional parameters for getting crypto bars.
```
Comment (kalmant) [inline marketdata/rest.go:1253]: small "typo" in this comment

File: marketdata/rest.go [hunk 3]
```diff
@@ -1234,6 +1239,108 @@ func (c *client) GetCryptoSnapshot(symbol string, exchange string) (*CryptoSnaps
...
+	Symbols []string
+	// Start is the inclusive beginning of the interval
+	Start time.Time
+	// End is the inclusive end of the interval
+	End time.Time
+	// Sort sets the sort order of the results. Sorting will be done by the UpdatedAt field.
+	Sort Sort
+	// IncludeContent tells the server to include the article content in the response.
+	IncludeContent bool
+	// ExcludeContentless tells the server to exclude articles that has no content.
```
Comment (kalmant) [inline marketdata/rest.go:1266]: "have"

File: marketdata/rest.go [hunk 4]
```diff
@@ -1234,6 +1239,108 @@ func (c *client) GetCryptoSnapshot(symbol string, exchange string) (*CryptoSnaps
...
+	ExcludeContentless bool
+	// TotalLimit is the limit of the total number of the returned news.
+	//
+	// If it's zero then the NoTotalLimit parameter is considered: if NoTotalLimit is true,
+	// then all the articles in the given start-end interval are returned.
+	// If NoTotalLimit is false, then 50 articles will be returned.
```
Comment (kalmant) [inline marketdata/rest.go:1272]: I wouldn't mention 50 here, I would talk about using the maximum for one API call.

File: marketdata/rest.go [hunk 5]
```diff
@@ -1234,6 +1239,108 @@ func (c *client) GetCryptoSnapshot(symbol string, exchange string) (*CryptoSnaps
...
+	if err != nil {
+		return nil, fmt.Errorf("invalid news url: %w", err)
+	}
+
+	q := u.Query()
+	setNewsQuery(q, params)
+	received := 0
+	totalLimit := params.TotalLimit
+	if params.TotalLimit == 0 && !params.NoTotalLimit {
+		totalLimit = 50
```
Comment (kalmant) [inline marketdata/rest.go:1317]: `newsMaxLimit` here?

File: marketdata/rest.go (hunk 4 above)
Comment (kalmant) [reply marketdata/rest.go:1272]: Or is it 50 on purpose and not the default?

File: marketdata/stream/entities.go [hunk 6]
```diff
@@ -147,6 +147,17 @@ type CryptoBar struct {
 	VWAP       float64
 }
 
+type News struct {
```
Comment (kalmant) [inline marketdata/stream/entities.go:150]: same question about ID

File: marketdata/stream/message.go [hunk 7]
```diff
@@ -626,6 +724,7 @@ var subMessageHandler = func(c *client, s subscriptions) error {
 	c.sub.lulds = s.lulds
 	c.sub.cancelErrors = s.cancelErrors
 	c.sub.corrections = s.corrections
+	c.sub.news = s.news
```
Comment (kalmant) [inline marketdata/stream/message.go:727]: I think you may have to make the corresponding change in  `handleSubscriptionMessage`.

File: marketdata/rest.go (hunk 4 above)
Comment (gnvk) [reply marketdata/rest.go:1272]: There's no trivial way to use the server-side page limit maximum here, that's why I opted for this solution instead.

**Review by kalmant — APPROVED** [review]
Thank you!

**Pull request merged**